File: PjblRa4/SorteadorIntervalo.java
import java.util.HashSet;
import java.util.Random;
import java.util.Set;

public class SorteadorIntervalo {
    private Random random;

    public SorteadorIntervalo() {
        random = new Random();
    }

    public Set<Integer> sortearIntervalo(int min, int max, int quantidade) {
        Set<Integer> resultados = new HashSet<>();
        if (min > max || quantidade <= 0 || quantidade > (max - min + 1)) {
            return null; 
        }

        while (resultados.size() < quantidade) {
            int numero = random.nextInt(max - min + 1) + min;
            resultados.add(numero);
        }

        return resultados;
    }
}

File: Ajuste/SorteadorIntervalo.py
import random  

class SorteadorIntervalo:
    def __init__(self):
        pass

    def sortear_intervalo(self, min_val, max_val, quantidade):
        if min_val > max_val or quantidade <= 0 or quantidade > (max_val - min_val + 1):
            return None

        resultados = set()
        while len(resultados) < quantidade:
            numero = random.randint(min_val, max_val)
            resultados.add(numero)

        return resultados